dataset_loaders/videos/davis2017.py

```diff
@@ -0,0 +1,279 @@
+                                   self.which_set + '.txt')) as f:
+                content = f.readlines()
+                f.close()
+            self._prefix_list = [prefix.strip() for prefix in content]
+        return self._prefix_list
+
+    @property
+    def unique_rgbs(self):
+        if self._unique_rgbs is None:
+            self._unique_rgbs = np.load(self.rgb_values_local_path).item()
+        return self._unique_rgbs
+
+    def save_rgbs(self):
+        unique_rgbs = {}
+        # Get first file name for this set and the corresponding
+        # list of unique rgb values
+        for root, dirs, files in os.walk(self.mask_path):
+            if files == []:
+                continue
+            mask_rgbs = []
+            name = sorted(files)[0]
+            # Unique list of rgbs
+            mask = Image.open(os.path.join(root, name)).convert('RGB')
+            # getcolors() returns a list of unique RGB values
+            # contained in an image. The elements in the list are
+            # in the form [num, RGB] where num is simply an
+            # incremental id of the unique RGB values
+            # For each element in the list take only the RGB value
+            for rgb in mask.getcolors():
+                mask_rgbs.append(rgb[1])
+            unique_rgbs[os.path.split(root)[1]] = sorted(mask_rgbs)
+        # Save the per-frame RGB values in a numpy file
+        np.save(self.rgb_values_shared_path, unique_rgbs)
+
+    @property
+    def filenames(self):
+        if self._filenames is None:
+            self._filenames = {}
+            # Get file names for this set
+            for vid_dir in self.prefix_list:
+                self._filenames[vid_dir] = []
+                for root, dirs, files in os.walk(os.path.join(
+                        self.image_path, vid_dir)):
+                    for name in files:
+                        self._filenames[vid_dir].append(os.path.join(
+                            os.path.split(root)[1], name[:-4]))
+                self._filenames[vid_dir].sort(key=natural_keys)
+        return self._filenames
+
+    def __init__(self,
+                 which_set='train',
+                 multiobject=True,
+                 dataset_version='2017',
+                 threshold_masks=False,
+                 split=.75,
+                 *args, **kwargs):
+
+        """ The Davis 2017 dataset
+        The following loader allows to load both the 2016 and 2017
+        version of the Davis dataset.
+
+        Parameters
+        ----------
+        which_set: string
+            A string in ['train', 'val', 'test-dev'], corresponding to
+            the training, validation and test sets specifically.
+        multiobject: boolean
+            In the case of the last dataset version (2017), this
+            parameter allows to decide when to return the labels in
+            multi-object notation (more then one object instance) or in the
+            binary one (only two ids 0-1).
```

Comment:
For the 2017 version of the dataset, determines whether to return the instance segmentation masks or the foreground-background ones.